ceph_deploy/util/pkg_managers.py

```diff
@@ -177,3 +177,93 @@ def zypper_refresh(conn):
         conn,
         cmd
     )
+
+
+class PackageManager(object):
+    """
+    Base class for all Package Managers
+    """
+
+    def __init__(self, remote_conn):
+        self.remote_info = remote_conn
+        self.remote_conn = remote_conn.conn
+
+    def _run(self, cmd):
+        return remoto.process.run(
+            self.remote_conn,
+            cmd
+        )
+
+class RPMManagerBase(PackageManager):
+    """
+    Base class to hold common pieces of Yum and DNF
+    """
+
+    def __init__(self, remote_conn, cmd):
+        super(RPMManagerBase, self).__init__(remote_conn)
+        self.cmd = cmd
+
+    def install(self, packages):
+        if isinstance(packages, str):
+            packages = [packages]
+
+        cmd = [
+            self.cmd,
+            '-y',
+            'install',
+        ]
+        cmd.extend(packages)
+        return self._run(cmd)
+
+    def remove(self, packages):
+        if isinstance(packages, str):
+            packages = [packages]
+
+        cmd = [
+            self.cmd,
+            '-y',
+            '-q',
+            'remove',
+        ]
+        cmd.extend(packages)
+        return self._run(cmd)
+
+    def clean(self, item=None):
+        item = item or 'all'
+        cmd = [
+            self.cmd,
+            'clean',
+            item,
+        ]
+
+        return self._run(cmd)
+
+
+class DNF(RPMManagerBase):
+    """
+    The DNF Package manager
+    """
+
+    def __init__(self, remote_conn):
+        super(DNF, self).__init__(remote_conn, 'dnf')
```

Comment:
I don't see the need to require passing `cmd` to `__init__` because the classes use `cmd` as well that is different from `self.cmd` and because this can just be defined as a class attribute:

    class DNF(RPMManagerBase):

        executable = 'dnf'

It doesn't need to be named `executable` but the idea is that this would skip the need to requiring it